sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1121,7 +1159,7 @@ def is_cmis_application_update_required(self, api, channel, speed):
         if speed == 0 or channel == 0 or api.is_flat_memory():
             return False
 
-        app_new = self.get_cmis_application_desired(api, channel, speed)
+        app_new = self.get_cmis_application_desired(api, host_lane_count, speed)
```

Comment:
1. current logic checks for all 8 host lanes whereas we should care for the requested host lanes only
2. Also we should make use of the selected/desired application during init state and compare it with the current active application
3. Also we should check for DP != Activated early in the function

Reply:
I have handled #2 now. As discussed, we are already handling #1 and #3 and have changed CMIS_NUM_CHANNELS to CMIS_MAX_HOST_LANES.